### PR TITLE
Fix update document with knnn_vector size not matching issue

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
This PR fixed the issue when updating a document with knn_vector field, the update fails because the vector dimension is not matching the required dimension.

### Issues Resolved
https://github.com/opensearch-project/neural-search/issues/207

### Check List
- [x] New functionality includes testing.
    - [x] All tests pass
- [x] New functionality has been documented.
    - [x] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by navneet1v** [comment]
@zane-neo before we review the PR fix the gradle checks.

**Comment by zane-neo** [comment]
> @zane-neo before PR can be reviewed, can you check why gradle checks are failing?

Added commons-lang3 back, in my local even I removed this dependency the project can be compiled successfully, and adding back this can cause jar hell issue when building neural-search plugin, but it seems have to add it back to pass the build.

**Comment by navneet1v** [comment]
@zane-neo the build and test gradle check is still failing. Please fix them.

**Comment by zane-neo** [comment]
> @zane-neo the build and test gradle check is still failing. Please fix them.

@navneet1v opensearch core changed some classes breaking the code build, I've created a PR to fix this issue in k-NN repo: https://github.com/opensearch-project/k-NN/pull/958, please help take a look. Thanks.

File: src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java [hunk 1]
```diff
@@ -119,7 +117,7 @@ void appendVectorFieldsToDocument(IngestDocument ingestDocument, Map<String, Obj
         Objects.requireNonNull(vectors, "embedding failed, inference returns null result!");
         log.debug("Text embedding result fetched, starting build vector output!");
         Map<String, Object> textEmbeddingResult = buildTextEmbeddingResult(knnMap, vectors, ingestDocument.getSourceAndMetadata());
-        textEmbeddingResult.forEach(ingestDocument::appendFieldValue);
+        textEmbeddingResult.forEach(ingestDocument::setFieldValue);
```
Comment (navneet1v) [inline src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java:122]: can we rename the function from appendVectorFieldsToDocument to setVectorFieldsToDocument, as we are not appending the values

**Review by navneet1v — APPROVED** [review]

**Comment by zane-neo** [comment]
Build is failing caused by opensearch core breaking change, and is fixed by k-NN team in this PR: https://github.com/opensearch-project/k-NN/pull/963.

**Review by ylwu-amzn — APPROVED** [review]

**Comment by junqiu-lei** [comment]
Looks like this PR directly merged to 2.x branch instead of main, I am adding 2.9 release tag for this one. @zane-neo @ylwu-amzn

**Comment by navneet1v** [comment]
@zane-neo this PR also needs to be raised on main branch.

Please going forward follow the below practice.
The first PR needs to be raised against the main branch first, then once it is approved it needs to be backported to 2.x.

**Pull request merged**